### ISSUE TITLE
Inheritance diagrams

### ISSUE THREAD

**Issue by corykinney** [mkdocstrings/python#83]
**Describe the solution you'd like**
It would be interesting to have the option to display a generated inheritance diagram with links to the documentation for parent/child classes. This could be a good option for a second Insiders funding tier. 

```mermaid
classDiagram
    class Parent
    link Parent "#module.Parent"
    Parent <|-- Child 
```

The mermaid.js `classDiagram` is designed for inheritance diagrams like this and even supports linking. I imagine it wouldn't be too hard to parse the inheritance structure and assemble the corresponding diagram syntax, for example:

```
classDiagram
    class Parent
    link Parent "#module.Parent"
    Parent <|-- Child 
```

The option could be off by default and toggled on specific classes like so:

```yaml
options:
  show_inheritance_diagram: True
```

**Comment by pawamoy** [comment]
Hi @corykinney, that's a cool idea! Thanks for the suggestion, I'll keep that in mind :slightly_smiling_face:

**Comment by pawamoy** [comment]
I have implemented this, see [the docs](https://mkdocstrings.github.io/python/usage/configuration/general/#show_inheritance_diagram). It's in the [$1500/month funding goal](http://localhost:8000/insiders/#1500-hyperlamp-navigation-tips).

For now, I used simple flowcharts, as I was afraid showing all attributes and all methods in a real Mermaid `classDiagram` by default would most of the times create too big diagrams, rendering them unreadable. We can totally reconsider and iterate an improved version later, of course (not as a new insiders feature, just as an enhancement :+1:).

Each node in the flowchart is clickable and will link to the corresponding class though :slightly_smiling_face: 
Feedback welcome!
